tests/testthat/test-assume.R

```diff
@@ -0,0 +1,310 @@
+    'though the argument `list\\(1\\)` was supplied'
+  )
+  
+  expect_error(
+    gss %>% 
+      specify(age ~ partyid) %>% 
+      hypothesize(null = "independence") %>%
+      assume("F", nrow(gss) - 1, 1, 2), 
+    'arguments `list\\(1, 2\\)` were supplied'
+  )
+  
+  # supply `distribution`s that don't align with the supplied variables
+  expect_error(
+    gss %>% 
+      specify(age ~ finrela) %>% 
+      hypothesize(null = "independence") %>%
+      assume("t", nrow(gss) - 1),
+    'supplied distribution "t" is not well-defined.*onse variable \\(age\\)'
+  )
+  
+  expect_error(
+    gss %>% 
+      specify(age ~ finrela) %>% 
+      hypothesize(null = "independence") %>%
+      assume("z", nrow(gss) - 1),
+    'supplied distribution "z" is not well-defined.*onse variable \\(age\\)'
+  )
+  
+  expect_error(
+    gss %>% 
+      specify(age ~ NULL) %>% 
+      hypothesize(null = "point", mu = 40) %>%
+      assume("z", nrow(gss) - 1),
+    'supplied distribution "z" is not well-defined.*onse variable \\(age\\)'
+  )
+  
+  # supply bad `x` arguments
+  expect_error(
+    gss %>% 
+      assume("z", nrow(gss) - 1),
+    '`x` argument must be the output of a core infer function'
+  )
+  
+  expect_error(
+    "boop" %>%
+      assume("z", nrow(gss) - 1),
+    '`x` argument must be the output of a core infer function'
+  )
+})
+
+test_that("assume() handles automatic df gracefully", {
+  expect_equal(
+    expect_silent(
+      gss %>%
+        specify(response = hours) %>%
+        hypothesize(null = "point", mu = 40) %>%
+        assume("t")
+    ),
+    expect_silent(
+      gss %>%
+        specify(response = hours) %>%
+        hypothesize(null = "point", mu = 40) %>%
+        assume("t")
+    )
+  )
+  
+  expect_message(
+    gss %>%
+      specify(response = hours) %>%
+      hypothesize(null = "point", mu = 40) %>%
+      assume("t", nrow(gss) - 2),
+    "does not match its expected value..*calculation for `df`"
+  )
+  
+  # t.test param with var.equal = FALSE
+  expect_silent(
+    gss %>%
+      specify(age ~ college) %>%
+      hypothesize(null = "independence") %>%
+      assume(distribution = "t", 423)
+  )
+  
+  # t.test param with var.equal = TRUE
+  expect_silent(
+    gss %>%
+      specify(age ~ college) %>%
+      hypothesize(null = "independence") %>%
+      assume(distribution = "t", 498)
+  )
+  
+  # min(n1 - 1, n2 - 1)
+  expect_silent(
+    gss %>%
+      specify(age ~ college) %>%
+      hypothesize(null = "independence") %>%
+      assume(distribution = "t", 173)
+  )
+  
+  # n1 + n2 - 2
+  expect_silent(
+    gss %>%
+      specify(age ~ college) %>%
+      hypothesize(null = "independence") %>%
+      assume(distribution = "t", 498)
+  )
+  
+  # supplied df persists if recognized but not default
+  expect_equal(
+    gss %>%
+      specify(age ~ college) %>%
+      hypothesize(null = "independence") %>%
+      assume(distribution = "t", 173) %>%
+      attr("df"),
+    173
+  )
+})
```

Comment:
Can you add similar tests for all dfs that are recognized but not default? I guess there's just one more of them.

I think that should be it for this PR so I'll approve now but would be good to add that test and make sure it passes before we merge.

Reply:
You got it!